### DISCUSSION TITLE
How a mediation recipient does make a connection with a mediator

### DISCUSSION THREAD

**Discussion by jakubkoci** [openwallet-foundation/credo-ts#429] [Architecture & Design]
We started the discussion in #250, but it deserves its own issue.

It seems there is no specification of how it should work. It's a question of whether it should be part of the framework. In any case, I would like to understand how to solve it :)

I like the idea with [did:web](https://w3c-ccg.github.io/did-method-web/) method and having `https://w3c-ccg.github.io/.well-known/did.json` on mediator server where an edge agent can find DID Document with service and keys. We don't even need to have [web-did-resolver](https://github.com/decentralized-identity/web-did-resolver) at the start.

But I'm not sure how to continue from there. 
* Should the edge agent send a connection invitation to the mediator? 
* Should it skip the connection invitation and send a connection request directly? 
* Should the edge agent get the invitation from the mediator in any other way?

**Comment by TimoGlastra** [comment]
When using a public did you can use it as an implicit invitation. So I'd say the second option:

> - Should it skip the connection invitation and send a connection request directly?

---

> We don't even need to have web-did-resolver at the start.

Specific reason not to use it? I'd like to add a did resolver in the near future that can use did resolvers from that library.

**Comment by TimoGlastra** [comment]
It provides some nice features with regards to did url parsing and caching etc.

https://github.com/decentralized-identity/did-resolver

Related issue: https://github.com/hyperledger/aries-framework-javascript/issues/172

**Comment by jakubkoci** [comment]
I'm not saying we shouldn't use it, just it's not necessary to have at least some improvement on the way forward.

**Comment by TimoGlastra** [comment]
Closing as completed
